src/Foundatio/Jobs/JobBase.cs

```diff
@@ -7,17 +7,20 @@
 
 namespace Foundatio.Jobs;
 
-public abstract class JobBase : IJob, IHaveLogger
+public abstract class JobBase : IJob, IHaveLogger, IHaveTimeProvider
 {
+    private readonly TimeProvider _timeProvider;
     protected readonly ILogger _logger;
 
-    public JobBase(ILoggerFactory loggerFactory = null)
+    public JobBase(TimeProvider timeProvider = null, ILoggerFactory loggerFactory = null)
```

Comment:
I really hate when loggerFactory is not last, this has the potential to break a bit or cause us to use positional arguments. Should we add a second constructor where logger factory is required?

Reply:
Added an second constructor